src/Stmt.h

```diff
@@ -606,9 +606,8 @@ class WhenInfo
 
 private:
 	// True if the "when" statement corresponds to old-style deprecated
-	// semantics (no captures, but needing captures).  Also generates
-	// the corresponding deprecation warnings, which are associated
-	// with "ws".
+	// semantics (no captures, but needing captures).  Also triggers
+	// an error associated with "ws".
 	bool IsDeprecatedSemantics(StmtPtr ws);
```

Comment:
Does the naming of this method make sense since it's not deprecated semantics anymore? Maybe it could be `IsInvalidSemantics`?

Reply:
Possibly, yes. There was this comment from @vpax :

> Regarding ripping out code, there's more that can be done for whens, happy to do so after this gets merged.

Vern might have thoughts what to call it down the road, too, or remove the helper from the class altogether. I'll leave it for now as is as it seems there's more to drop. Leaving it as is for now.